src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -384,6 +384,19 @@ public ProgrammingSubmission addProgrammingSubmission(ProgrammingExercise exerci
         return submission;
     }
 
+    @Transactional
+    public ProgrammingSubmission addProgrammingSubmissionWithResult(ProgrammingExercise exercise, ProgrammingSubmission submission, String login) {
```

Comment:
I think it would be nice to have some JavaDoc on the util methods of common test services like the `DatabaseUtilService` here. What do you think @thilo-behnke ?

Reply:
I added some docs, I agree that the test utils are getting more complicated.